client/src/_actions/user.actions.js

```diff
@@ -1,45 +1,78 @@
 import {
-  userConstants,
+    userConstants,
 } from '../_constants'
 import {
-  instance,
+    instance,
 } from '../axios.instanse'
+import authToken from '../utils/AuthToken'
+import jwt_decode from 'jwt-decode'
+import { _setAuthData } from '../utils/auth.service'
 
 export const userActions = {
-  register,
+    register,
+    login
 }
 
 function register(user) {
-  return dispatch => {
-    dispatch(registrationRequest(user))
-    instance.post('/register', user)
-      .then(
-        () => {
-          dispatch(registrationSuccess())
+    return dispatch => {
+        dispatch(registrationRequest(user))
+        instance.post('/register', user)
+            .then(
+                () => {
+                    dispatch(registrationSuccess())
+                }
+            ).catch(error => {
+            dispatch(registrationFailure(error.response.data && error.response.data.message))
+        })
+    };
+
+    function registrationRequest(user) {
+        return {
+            type: userConstants.REGISTER_REQUEST,
+            user,
         }
-      ).catch(error => {
-        dispatch(registrationFailure(error.response.data && error.response.data.message))
-      })
-  }
+    }
 
-  function registrationRequest(user) {
-    return {
-      type: userConstants.REGISTER_REQUEST,
-      user,
+    function registrationSuccess(user) {
+        return {
+            type: userConstants.REGISTER_SUCCESS,
+            user,
+        }
     }
-  }
 
-  function registrationSuccess(user) {
-    return {
-      type: userConstants.REGISTER_SUCCESS,
-      user,
+    function registrationFailure(error) {
+        return {
+            type: userConstants.REGISTER_FAILURE,
+            error,
+        }
     }
-  }
+}
 
-  function registrationFailure(error) {
-    return {
-      type: userConstants.REGISTER_FAILURE,
-      error,
+
+function login(user) {
+    return dispatch => {
+        instance.post('/login', user)
+            .then(res => {
+                const { accessToken } = res.data;
+                authToken( accessToken );
+                _setAuthData(res);
```

Comment:
why do you add so many tokens and expiresIn after login?
one token is not enough?
localStorage.setItem('refreshToken', response.data.refreshToken);
    localStorage.setItem('accessToken', response.data.accessToken);
    localStorage.setItem('expiresIn', response.data.expires);